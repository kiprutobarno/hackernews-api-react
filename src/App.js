import React, { Component } from "react";

const list = [
  {
    title: "React",
    url: "https://facebook.github.io/react",
    author: "Jordan Walke",
    num_comments: 3,
    points: 4,
    objectID: 0
  },
  {
    title: "Redux",
    url: "https://github.com/reactjs/redux",
    author: "Dan Abramov, Andrew Clark",
    num_comments: 2,
    points: 5,
    objectID: 1
  }
];

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { list };
  }
  render() {
    return (
      <div className="App">
        <table>
          <thead>
            <th>ID</th>
            <th>Title</th>
            <th>Author</th>
            <th>Comments</th>
            <th>Points</th>
          </thead>
          <tbody>
            {this.state.list.map(item => {
              return (
                <tr key={item.objectID}>
                  <td>{item.objectID}</td>
                  <td>{item.title}</td>
                  <td>{item.author}</td>
                  <td>{item.num_comments}</td>
                  <td>{item.points}</td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  }
}

export default App;
