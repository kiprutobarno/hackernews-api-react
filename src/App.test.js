import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App, { Search, Button, Table } from "./App";

describe("App", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //implement snapshot test
  test("has valid snapshot", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Search", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Search>Search</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test("has a valid snapshot", () => {
    const component = renderer.create(<Search>Search</Search>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Button", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button>Give me more</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test("has a valid snapshot", () => {
    const component = renderer.create(<Button>Give me more</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Table", () => {
  const props = {
    list: [
      { title: "Metaphysics", author: "Barno", num_comments: 6, objectID: "B" },
      {
        title: "Biosystems",
        author: "Kimberly",
        num_comments: 2,
        objectID: "X"
      }
    ]
  };

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Table {...props} />, div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Table {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
