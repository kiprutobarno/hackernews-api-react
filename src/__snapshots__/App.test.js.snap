// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has valid snapshot 1`] = `
<div
  className="App"
>
  <div
    className="interactions"
  >
    <div
      className="form"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>
  <div
    className="interactions"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      More
    </button>
  </div>
</div>
`;

exports[`Button has a valid snapshot 1`] = `
<button
  type="button"
>
  Give me more
</button>
`;

exports[`Search has a valid snapshot 1`] = `
<div
  className="form"
>
  <form>
    <input
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
